mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -919,6 +948,10 @@ class LogisticRegressionModel private[spark] (
     }
   }
 
+  private[classification] val probability2predictionUdf = udf {
```

Comment:
This UDF might pull this entire class into the closure and thus into the summaries.  I believe we can eliminate this UDF; I'll comment more below.